core/src/components/calendar/calendar.tsx

```diff
@@ -0,0 +1,555 @@
+    e.stopPropagation();
+    // Stop the click from being propagated after dragging/mouseup.
+    window.removeEventListener('click', this.captureClick, true);
+  }
+
+  isDateDisabled(date: Date) {
+    if (this.disabled) {
+      return true;
+    }
+    let disabled = false;
+    if (this.availableDates && this.availableDates.length > 0) {
+      disabled = !this.isDateInArray(date, this.availableDates);
+    }
+    if (!disabled && this.disabledDates && this.disabledDates.length > 0) {
+      disabled = this.isDateInArray(date, this.disabledDates);
+    }
+    if (!disabled && this.disabledBeforeDate) {
+      disabled = date < this.disabledBeforeDate;
+    }
+    if (!disabled && this.disabledAfterDate) {
+      disabled = date > this.disabledAfterDate;
+    }
+    if (!disabled && this.disabledDays && this.disabledDays.length > 0) {
+      const weekday = date.getDay() as 0 | 1 | 2 | 3 | 4 | 5 | 6;
+      disabled = this.disabledDays.indexOf(weekday) !== -1;
+    }
+    return disabled;
+  }
+
+  getDayView(day: Date, month: -1 | 0 | 1) {
+    const disabled = this.isDateDisabled(day);
+    return (
+      <div
+        class={{
+          'day-outer': true,
+          'day-selected': this.isDateSelected(day),
+          today: this.isToday(day),
+          'not-current-month': !this.isCurrentMonth(day, month),
+          'disabled': disabled
+        }}
+      >
+        <div>
+          <div onClick={() => disabled ? null : this.selectDate(day)} class="day-inner">
+            {day.getDate()}
+            <ion-ripple-effect />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    const mode = getIonMode(this);
+    return (
+      <Host
+        onMouseUp={this.handleMouseUp}
+        class={{
+          [mode]: true,
+          // Used internally for styling
+          [`calendar-${mode}`]: true
+        }}
+      >
+        <div style={{ overflow: 'hidden' }}>
+          <div class="months-wrapper">
+            {([-1, 0, 1] as (1 | 0 | -1)[]).map(month => (
+              <div
+                class={{
+                  'month-wrapper': true,
+                  'previous-month': month === -1,
+                  'current-month': month === 0,
+                  'next-month': month === 1
+                }}
+              >
+                <div class="calendar-header">
+                  <div class="calendar-title-wrapper">
+                    <ion-button
+                      class="month-title"
+                      fill="clear"
+                      onClick={() => this.toggleYearView()}
+                    >
+                      {this.getMonthTitle(month)}
+                      <ion-icon
+                        size="small"
+                        name={this.showYears ? 'caret-up' : 'caret-down'}
+                        slot="end"
+                      />
+                    </ion-button>
+                  </div>
+                  <div />
+                  <ion-button
```

Comment:
Can this be a native button? I can help with this. We usually don't use `ion-button` when building components because it pulls in all of the button CSS to this component. See `ion-menu-button`: https://github.com/ionic-team/ionic/blob/master/core/src/components/menu-button/menu-button.tsx#L76-L105